### PR TITLE
static: update with latest changes from moby

### PR DESCRIPTION
**Pull request by crazy-max**
follow-up https://github.com/moby/moby/pull/44546
closes #801 

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

File: static/Makefile [hunk 1]
```diff
@@ -9,20 +9,17 @@ HASH_CMD=docker run -v $(CURDIR):/sum -w /sum debian:jessie bash hash_files
 DIR_TO_HASH:=build/linux
 DOCKER_CLI_GOLANG_IMG=golang:$(GO_VERSION)
 
-DOCKER_BUILD_OPTS=
+BAKE_SET_FLAGS=
```
Comment (thaJeztah) [inline static/Makefile:12]: The only thing I had with this was that I was wondering if someone may be depending on this variable; wondering if there's a way to keep it for backward-compatibility (might be tricky as it needs a different format for `bake`).

Comment (crazy-max) [reply static/Makefile:12]: I think we can keep the build args we have previously, let me take a look.

File: static/Makefile [hunk 2]
```diff
@@ -13,18 +15,17 @@ DOCKER_BUILD_OPTS=
 
 ifneq ($(strip $(CONTAINERD_VERSION)),)
 # Set custom build-args to override the containerd version to build for static
-# packages. The Dockerfile for 20.10 and earlier used CONTAINERD_COMMIT, later
-# versions use CONTAINERD_VERSION. We can remove CONTAINERD_VERSION once 20.10.x
-# reaches EOL.
+# packages.
 DOCKER_BUILD_OPTS +=--build-arg=CONTAINERD_VERSION=$(CONTAINERD_VERSION)
-DOCKER_BUILD_OPTS +=--build-arg=CONTAINERD_COMMIT=$(CONTAINERD_VERSION)
```
Comment (neersighted) [inline static/Makefile:20]: For other reviews: this comment is out of date; 20.10 now uses `CONTAINERD_VERSION`.

File: static/Makefile [hunk 3]
```diff
@@ -4,6 +4,8 @@ CLI_DIR=$(realpath $(CURDIR)/../src/github.com/docker/cli)
 ENGINE_DIR=$(realpath $(CURDIR)/../src/github.com/docker/docker)
 BUILDX_DIR=$(realpath $(CURDIR)/../src/github.com/docker/buildx)
 
+ENGINE_GITCOMMIT?=$(shell cd $(realpath $(CURDIR)/../src/github.com/docker/docker) && git rev-parse --short HEAD)
```
Comment (thaJeztah) [inline static/Makefile:7]: Was wondering; is this different from `ENGINE_REF`? https://github.com/docker/docker-ce-packaging/blob/47167619d0386bee288924b58aad5f71daf1e13f/common.mk#L41

Comment (crazy-max) [reply static/Makefile:7]: Same as https://github.com/docker/docker-ce-packaging/blob/47167619d0386bee288924b58aad5f71daf1e13f/deb/Makefile#L6

Comment (thaJeztah) [reply static/Makefile:7]: Ah! Yeah, we should look if we can share more of those (probably the complication was to define the correct path)

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Review by neersighted — APPROVED** [review]

**Pull request merged**